conda_smithy/tests/test_feedstock_io.py

```diff
@@ -0,0 +1,710 @@
+            fh.write(write_text)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual(write_text, read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual(write_text, read_text)
+
+
+    def test_touch_file_1(self):
+        filename = "test.txt"
+        filename = os.path.join(self.tmp_dir, filename)
+
+        fio.touch_file(filename)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual("", read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual("", read_text)
+
+
+    def test_touch_file_2(self):
+        filename = "test.txt"
+
+        fio.touch_file(filename)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual("", read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual("", read_text)
+
+
+    def test_touch_file_3(self):
+        filename = "dir1/dir2/test.txt"
+
+        fio.touch_file(filename)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual("", read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual("", read_text)
+
+
+    def test_remove_file_1(self):
+        filename = "test.txt"
+        filename = os.path.join(self.tmp_dir, filename)
+
+        with open(filename, "w") as fh:
+            fh.write("")
+
+        self.repo.index.add([filename])
+
+        self.assertTrue(os.path.exists(filename))
+        self.assertTrue(list(self.repo.index.iter_blobs(BlobFilter(filename))))
+
+        fio.remove_file(filename)
+
+        self.assertFalse(os.path.exists(filename))
+        self.assertFalse(list(self.repo.index.iter_blobs(BlobFilter(filename))))
+
+
+    def test_remove_file_2(self):
+        filename = "test.txt"
+
+        with open(filename, "w") as fh:
+            fh.write("")
+
+        self.repo.index.add([filename])
+
+        self.assertTrue(os.path.exists(filename))
+        self.assertTrue(list(self.repo.index.iter_blobs(BlobFilter(filename))))
+
+        fio.remove_file(filename)
+
+        self.assertFalse(os.path.exists(filename))
+        self.assertFalse(list(self.repo.index.iter_blobs(BlobFilter(filename))))
+
+
+    def test_remove_file_3(self):
+        filename = "dir1/dir2/test.txt"
+        dirname = os.path.dirname(filename)
+
+        if dirname and not os.path.exists(dirname):
```

Comment:
Small nit: this line is not needed, the directory will never exist because `setUp` creates a fresh directory